Add content for HIPRTC compilation cache (#3733)
ROCm-Developer-Tools/HIP: docs/how-to/hip_rtc.md

HIP lets you compile kernels at runtime with the `hiprtc*` APIs.
Kernels can be stored as a text string and can be passed to HIPRTC APIs alongside options to guide the compilation.

:::{note} 

>  * This library can be used on systems without HIP installed nor AMD GPU driver installed at all (offline compilation). Therefore, it does not depend on any HIP runtime library.
>  * But it does depend on Code Object Manager (comgr). You may try to statically link comgr into HIPRTC to avoid any ambiguity.
>  * Developers can decide to bundle this library with their application.

Reviewer: ```suggestion
 * This library can be used on systems without HIP installed nor AMD GPU driver installed at all (offline compilation). Therefore, it doesn't depend on any HIP runtime library.
 * This library depends on Code Object Manager (comgr). You can try to statically link comgr into HIPRTC to avoid ambiguity.
 * Developers can bundle this library with their application.
```

Reviewer: I always have a problem with "nor" - never quite sure unless with "neither"

}
```

## Kernel Compilation Cache

HIPRTC incorporates a cache to avoid recompiling kernels between program executions. The contents of the cache include the kernel source code (including the contents of any `#include` headers), the compilation flags, and the compiler version. After a ROCm version update, the kernels are progressively recompiled, and the new results are cached. When the cache is disabled, each kernel is recompiled every time it is requested. 

> The cache's status as enabled or disabled, the location for storing the cache contents, and the cache eviction policy can be managed using the following environment variables: 

Reviewer: ```suggestion
Use the following environment variables to manage the cache's status as enabled or disabled, the location for storing the cache contents, and the cache eviction policy: 
```

Reviewer: Not sure that is tighter

> * `AMD_COMGR_CACHE_DIR`: By default the value of this environment variable is defined as `$XDG_CACHE_HOME/comgr_cache`, which defaults to `$USER/.cache/comgr_cache` on Linux, and `%LOCALAPPDATA%\cache\comgr_cache` on Windows. You can disable the cache by setting the environment variable to an empty string, "". You can also specify a different directory for the environment variable to change the path for cache storage. If the runtime fails to access the specified cache directory, the cache is disabled. 

Reviewer: ```suggestion
* `AMD_COMGR_CACHE_DIR`: By default, the value of this environment variable is defined as `$XDG_CACHE_HOME/comgr_cache`, which defaults to `$USER/.cache/comgr_cache` on Linux, and `%LOCALAPPDATA%\cache\comgr_cache` on Microsoft Windows. You can disable the cache by setting the environment variable to an empty string, "". You can also specify a different directory for the environment variable to change the path for cache storage. If the runtime fails to access the specified cache directory, the cache is disabled. 
```

> 

Reviewer 2: ```suggestion

By default, the cache is turned off, set the environment variable
`AMD_COMGR_CACHE=1` to enable it. This behavior may change in a future release.

* `AMD_COMGR_CACHE`: When set to a value different from 0, the cache is enabled.
```

Author: I added this separately.
